src/develop/imageop.c

```diff
@@ -1723,6 +1779,19 @@ void dt_iop_commit_params(dt_iop_module_t *module, dt_iop_params_t *params,
 
   module->commit_params(module, params, pipe, piece);
 
+  // adjust the label to match presets if possible or otherwise the default
+  // multi_name for this module.
+
+  if(!dt_iop_is_hidden(module)
+     && module_is_enabled
+     && module_params_changed
+     && !module->multi_name_hand_edited)
+  {
+    if(module->label_recompute_handle)
+      g_source_remove(module->label_recompute_handle);
+    module->label_recompute_handle = g_timeout_add(500, _iop_update_label, module);
```

Comment:
By the time this gets called, the module (instance) may have been deleted (exiting darkroom, moving to next image).

Reply:
Indeed, the timer needs to be cleared when changing images.